cognitive/src/main/scala/com/microsoft/azure/synapse/ml/cognitive/MultivariateAnomalyDetection.scala

```diff
@@ -0,0 +1,396 @@
+  val source = new ServiceParam[String](this, "source", "The blob link to the input data. " +
+    "It should be a zipped folder containing csv files. Each csv file should has two columns with header 'timestamp'" +
+    " and 'value' (case sensitive). The file name will be used as the variable name. The variables used for" +
+    " detection should be exactly the same as for training. Please refer to the sample data to prepare your" +
+    " own data accordingly.", isRequired = true)
+
+  def setSource(v: String): this.type = setScalarParam(source, v)
+
+  def setSourceCol(v: String): this.type = setVectorParam(source, v)
+
+  def getSource: String = getScalarParam(source)
+
+  def getSourceCol: String = getVectorParam(source)
+}
+
+trait HasMADStartTime extends HasServiceParams {
+  val startTime = new ServiceParam[String](this, "startTime", "A required field, start time" +
+    " of data to be used for detection/generating multivariate anomaly detection model, should be date-time.",
+    isRequired = true)
+
+  def setStartTime(v: String): this.type = setScalarParam(startTime, v)
+
+  def setStartTimeCol(v: String): this.type = setVectorParam(startTime, v)
+
+  def getStartTime: String = getScalarParam(startTime)
+
+  def getStartTimeCol: String = getVectorParam(startTime)
+
+}
+
+trait HasMADEndTime extends HasServiceParams {
+  val endTime = new ServiceParam[String](this, "endTime", "A required field, end time of data" +
+    " to be used for detection/generating multivariate anomaly detection model, should be date-time.",
+    isRequired = true)
+
+  def setEndTime(v: String): this.type = setScalarParam(endTime, v)
+
+  def setEndTimeCol(v: String): this.type = setVectorParam(endTime, v)
+
+  def getEndTime: String = getScalarParam(endTime)
+
+  def getEndTimeCol: String = getVectorParam(endTime)
+}
+
+trait MADBase extends HasAsyncReply with HasMADSource with HasMADStartTime with HasMADEndTime
+  with HasCognitiveServiceInput with HasInternalJsonOutputParser with HasSetLocation with Wrappable
+  with HTTPParams with HasOutputCol with HasURL with ComplexParamsWritable
+  with HasSubscriptionKey with HasErrorCol with BasicLogging {
+
+  setDefault(
+    outputCol -> (this.uid + "_output"),
+    errorCol -> (this.uid + "_error"))
+
+  protected def queryForResult(key: Option[String],
+                               client: CloseableHttpClient,
+                               location: URI): Option[HTTPResponseData] = {
+    val get = new HttpGet()
+    get.setURI(location)
+    key.foreach(get.setHeader("Ocp-Apim-Subscription-Key", _))
+    get.setHeader("User-Agent", s"synapseml/${BuildInfo.version}${HeaderValues.PlatformInfo}")
+    val resp = convertAndClose(sendWithRetries(client, get, getBackoffs))
+    get.releaseConnection()
+    val fields = IOUtils.toString(resp.entity.get.content, "UTF-8").parseJson.asJsObject.fields
+    var status = "none"
+    if (fields.keySet.contains("modelInfo")) {
```

Comment:
var can be replaced with conditional equality. Ex:  val foo = if(..){1}else{2}

Reply:
Done! Good to know this!